### PR TITLE
Entity_Lecture1_Class

### PR DESCRIPTION
**Pull request by wi9916**
Start

### CODE REVIEW

File: Entity/Entity/Interfaces/Categoriesable.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Entity.Interfaces
+{
+    public interface Categoriesable
```
Comment (yevhenbilushko) [inline Entity/Entity/Interfaces/Categoriesable.cs:9]: create interfaces it is good for logic
this task is about DB structure
so no need interfaces
just DB structures
model with relations

Comment (yevhenbilushko) [reply Entity/Entity/Interfaces/Categoriesable.cs:9]: so rewrite please
remember - each model should have Id

File: Entity/Entity/Models/Categories.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿using Entity.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Entity.Models
+{
+    public class Categories: Categoriesable
```
Comment (yevhenbilushko) [inline Entity/Entity/Models/Categories.cs]: lets simplify 
i mean let it be only one level of Category

File: Entity/Entity/Interfaces/Customerable.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Entity.Interfaces
+{
+    public interface Customerable: Personable
```
Comment (yevhenbilushko) [inline Entity/Entity/Interfaces/Customerable.cs:9]: if it Customer
seems like it should have Name Phone Id

File: Entity/Entity/Interfaces/Deliverable.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Entity.Interfaces
+{
+    public interface Deliverable
+    {
```
Comment (yevhenbilushko) [inline Entity/Entity/Interfaces/Deliverable.cs:10]: do Deliverable and DeliveryManable related ?
seems like should be

File: Entity/Entity/Interfaces/Orderable.cs [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    public interface Orderable
+    {
+        int Id { get; set; }
+        DateTime CreatedDate { get; set; }
+        DateTime DeliveryDate { get; set; }
+        string AddressOfDelivery { get; set; }
+        int CustomerId { get; set; }
+        int DeliveryId { get; set; }
+        OrderStatus Status { get; set; }
+        IEnumerable<int> ProductsId { get; set; }
```
Comment (yevhenbilushko) [inline Entity/Entity/Interfaces/Orderable.cs:19]: better
IEnumerable<Product> Products { get; set; }

File: Entity/Entity/Interfaces/Customerable.cs (hunk 3 above)
Comment (wi9916) [reply Entity/Entity/Interfaces/Customerable.cs:9]: Yes he has. Because his parent is person. And person has all of it.

**Comment by wi9916** [comment]
I cut interfaces and make same fix

**Review by yevhenbilushko — APPROVED** [review]

**Pull request merged**